### ISSUE TITLE
Cannot serialize Date property correctly.

### ISSUE THREAD

**Issue by kenkiku1021** [lhorie/mithril.js#1179]
### Description:

When serializing a Date property object into JSON, an empty object created.
In m.request() function, a model object will serialize by JSON.stringify() in default.
The toJSON() function of the property object returns raw Date object, and it seems that JSON.stringify() doesn't expect it. So, the property is serialized into empty object.

### Steps to Reproduce:

```
var p = m.prop(new Date());
p(); // -> Date object (Thu Jul 28 2016 20:47:18 GMT+0900 (JST))
p.toJSON(); // -> 2016-07-28T11:47:18.896Z
JSON.stringify(p); // -> {} (empty object!)
```

### Expected:

I think a Date property object should be serialized by ISO8601 form.
In case of a property has Date object, toJSON() function should return Date.toISOString() value.

### Suggestion:

It is dirty but it works correctly.
```
function gettersetter(store) {
	function prop() {
		if (arguments.length) store = arguments[0]
		return store
	}

	prop.toJSON = function () {
		return store instanceof Date ? store.toISOString() : store // <- HERE
	}

	return prop
}
```

**Comment by isiahmeadows** [comment]
Better suggestion: check for `.toJSON` and call one if it exists. Something like this:

```js
function gettersetter(store) {
    function prop() {
        if (arguments.length) store = arguments[0]
        return store
    }

    prop.toJSON = function () {
        var method = store != null && store.toJSON
        return typeof method === "function" ? method.call(store) : store
    }

    return prop
}
```

**Comment by kenkiku1021** [comment]
Thank you for your suggestion.
It works fine.

**Comment by kenkiku1021** [comment]
I read other issues and pull requests again.
It seems that my problem is similar to the pull request [#1071](https://github.com/lhorie/mithril.js/pull/1071).

**Comment by isiahmeadows** [comment]
Yeah...that's rather dead ATM.

**Comment by lhorie** [comment]
merged #1071 and fixed in rewrite
